### PR TITLE
[new release] irmin project (3.5.1)

### PR DESCRIPTION
**Pull request by metanivek**
Project page: <a href="https://github.com/mirage/irmin">https://github.com/mirage/irmin</a>

### Fixed

- **irmin-pack**
  - Integrity check of a commit works on stores using the minimal indexing
    strategy. (mirage/irmin#2160, @icristescu)

### CODE REVIEW

File: packages/irmin/irmin.3.5.1/opam [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+  "uri"     {>= "1.3.12"}
+  "uutf"
+  "jsonm"   {>= "1.0.0"}
+  "lwt"     {>= "5.3.0"}
+  "digestif" {>= "0.9.0"}
+  "ocamlgraph"
+  "logs"    {>= "0.5.0"}
+  "bheap" {>= "2.0.0"}
+  "astring"
+  "mtime" {>= "1.0.0"}
```
Comment (raphael-proust) [inline packages/irmin/irmin.3.5.1/opam:30]:
```suggestion
  "mtime" {>= "1.0.0" && < "1.5.0"}
```

```
### output ###
# (cd _build/default && /home/opam/.opam/4.14/bin/ocamlc.opt -w -40 -g -bin-annot -I src/irmin/.irmin.objs/byte -I /home/opam/.opam/4.14/lib/angstrom -I /home/opam/.opam/4.14/lib/astring -I /home/opam/.opam/4.14/lib/base64 -I /home/opam/.opam/4.14/lib/bheap -I /home/opam/.opam/4.14/lib/bigstringaf -I /home/opam/.opam/4.14/lib/digestif -I /home/opam/.opam/4.14/lib/either -I /home/opam/.opam/4.14/lib/eqaf -I /home/opam/.opam/4.14/lib/fmt -I /home/opam/.opam/4.14/lib/jsonm -I /home/opam/.opam/4.14/lib/logs -I /home/opam/.opam/4.14/lib/lwt -I /home/opam/.opam/4.14/lib/mtime -I /home/opam/.opam/4.14/lib/ocamlgraph -I /home/opam/.opam/4.14/lib/optint -I /home/opam/.opam/4.14/lib/ppx_irmin/internal-lib -I /home/opam/.opam/4.14/lib/repr -I /home/opam/.opam/4.14/lib/stdlib-shims -I /home/opam/.opam/4.14/lib/stringext -I /home/opam/.opam/4.14/lib/uchar -I /home/opam/.opam/4.14/lib/uri -I /home/opam/.opam/4.14/lib/uutf -I src/irmin/data/.irmin_data.objs/byte -intf-suffix .ml -no-alias-deps -open Irmin__ -o src/irmin/.irmin.objs/byte/irmin__Logging.cmo -c -impl src/irmin/logging.pp.ml)
# File "src/irmin/logging.ml", line 39, characters 15-31:
# 39 |       let dt = Mtime.Span.to_us (Clock.count start_time) in
#                     ^^^^^^^^^^^^^^^^
# Error: Unbound value Mtime.Span.to_us
```

See release notes: https://github.com/dbuenzli/mtime/blob/master/CHANGES.md

Comment (raphael-proust) [reply packages/irmin/irmin.3.5.1/opam:30]: If this is the right fix and it gets applied, don't forget to upstream the it.

Comment (metanivek) [reply packages/irmin/irmin.3.5.1/opam]: thanks @raphael-proust -- I actually fixed this with the 3.5.0 release and upstreamed but then forgot to include it when I branched for this patch release. :facepalm:

**Comment by raphael-proust** [comment]
The CI errors seem to fall in the following categories:
- Timeouts (I've restarted those)
- Network errors when installing system packages (same)
- `ld`  exists with `1` on the more exotic architectures
- `ca-cert` issue on macos protected build env

I don't think those errors should block this PR

**Comment by kit-ty-kate** [comment]
* The macos errors should be restricted at the opam file level (disable the tests on macos)
* The mtime link failures seem related too. Could you have a look at them?

**Comment by metanivek** [comment]
@kit-ty-kate 

I disabled tests for macos with the packages that are failing. CI is running again but these errors should disappear.  The `libirmin` build is also failing `macos` but I did not remove its availability on that platform. This package does work on macos but unfortunately suffers from this known issue on opam CI: https://github.com/mirage/ca-certs/issues/20.

As for `mtime`, I can't quite figure this one out. Is this an `mtime` issue or an `irmin` one? All the linking failures are happening on "exotic" (as @raphael-proust puts it :smile:) architectures, but also all the failures are when building with OCaml 5. I'm happy to adjust something within Irmin but not sure what I would do (if there is indeed anything I can).

Thanks for the reviews @kit-ty-kate and @raphael-proust.

**Comment by samoht** [comment]
I don't understand what are the mtime failures. Merging as is as most of the "base" tests pass fine, but happy to take follow fixes.

**Pull request merged**